scanner/metrics.go

```diff
@@ -0,0 +1,84 @@
+package scanner
+
+import (
+	"context"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigenmetrics "github.com/Layr-Labs/eigensdk-go/metrics"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// Namespace
+const (
+	Namespace = "scanner"
+)
+
+// Metrics struct
+type Metrics struct {
+	logger logging.Logger
+
+	// Accumulated number of scan requests received.
+	AccNumRequests *prometheus.CounterVec
+	// The latency (in ms) to process the request.
+	RequestLatency *prometheus.SummaryVec
+	// EigenMetrics
+	EigenMetrics   eigenmetrics.Metrics
+
+	registry *prometheus.Registry
+	// socketAddr is the address at which the metrics server will be listening.
+	// should be in format ip:port
+	socketAddr string
+}
+
+// NewMetrics metrics
+func NewMetrics(eigenMetrics eigenmetrics.Metrics, reg *prometheus.Registry, logger logging.Logger, socketAddr string) *Metrics {
+
+	// Add Go module collectors
+	reg.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+	reg.MustRegister(collectors.NewGoCollector())
+
+	metrics := &Metrics{
+		// The "status" label has values: success, failure.
+		AccNumRequests: promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: Namespace,
+				Name:      "eigenda_scanner_requests_total",
```

Comment:
Make the namespace: `eigenda_scanner` otherwise I think the metric will be `scanner_eigenda_scanner_requests_total`